src/v/pandaproxy/schema_registry/storage.h

```diff
@@ -21,4 +24,123 @@ namespace pandaproxy::schema_registry {
 
 using topic_key_magic = named_type<int32_t, struct magic_tag>;
 
+struct schema_key {
+    ss::sstring keytype{"SCHEMA"};
```

Comment:
> is there a reason why key type doesn't get a strong type but the other fields do?

This is a little under-developed. I'm not yet sure how I want to deal with the various key types; `SCHEMA`, `CONFIG`, `MODE`, in terms of deserialisation from JSON, and since this is really the deserialisation layer, some type erasure needs to be performed. It might be better named as just `key`. Then some typing will be added for the purpose of applying the other key types to the store, possibly with a std::variant and visitation.

Reply:
got it. that's reasonable

Reply:
I made this a strong type.